integration-tests/gradle/src/test/java/io/quarkus/gradle/UberJarFormatWorksTest.java

```diff
@@ -0,0 +1,85 @@
+
+import io.quarkus.test.devmode.util.DevModeTestUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+import org.awaitility.core.ConditionTimeoutException;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static io.quarkus.gradle.LaunchUtils.launch;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class UberJarFormatWorksTest extends QuarkusGradleWrapperTestBase{
+
+    private static Future<?> jarRun;
+
+    @Test
+    public void testUberJarFormatWorks() throws Exception {
+
+        final File projectDir = getProjectDir("test-uber-jar-format-works");
+        runGradleWrapper(projectDir, "clean", "build");
+        final Path quarkusApp = projectDir.toPath().resolve("build");
+        assertThat(quarkusApp).exists();
+        Path jar = quarkusApp.resolve("code-with-quarkus-1.0.0-SNAPSHOT-runner.jar");
+        assertThat(jar).exists();
+
+        File output = new File(projectDir, "build/output.log");
+        output.createNewFile();
+        Process process = launch(jar, output);
+        try {
+            // Wait until server up
+            dumpFileContentOnFailure(() -> {
+                await()
+                        .pollDelay(1, TimeUnit.SECONDS)
+                        .atMost(1, TimeUnit.MINUTES).until(() -> DevModeTestUtils.getHttpResponse("/hello", 200));
+                return null;
+            }, output, ConditionTimeoutException.class);
+
+            String logs = FileUtils.readFileToString(output, "UTF-8");
+
+            assertThatOutputWorksCorrectly(logs);
+
+            // test that the application name and version are properly set
+            assertThat(DevModeTestUtils.getHttpResponse("/hello", getQuarkusDevBrokenReason()).equals("hello"));
+        } finally {
+            process.destroy();
+        }
+
+    }
+
+    static void assertThatOutputWorksCorrectly(String logs) {
+        assertThat(logs.isEmpty()).isFalse();
+        String infoLogLevel = "INFO";
+        assertThat(logs.contains(infoLogLevel)).isTrue();
+        assertThat(logs.contains("cdi, resteasy")).isTrue();
+    }
+
+    private void dumpFileContentOnFailure(final Callable<Void> operation, final File logFile,
+                                          final Class<? extends Throwable> failureType) throws Exception {
+
+        final Logger log = Logger.getLogger(FastJarFormatWorksTest.class);
+        try {
+            operation.call();
+        } catch (Throwable t) {
+            log.error("Dumping logs that were generated in " + logFile + " for an operation that resulted in "
+                    + t.getClass().getName() + ":", t);
+
+            throw t;
+        }
+    }
+
+    private static Supplier<String> getQuarkusDevBrokenReason() {
```

Comment:
The name of the method is incorrect since it has nothing to do with quarkus dev

Reply:
What should I use then ?

Reply:
It doesn't even need to be a method - it can be "inlined" into the call-site (` assertThat(DevModeTestUtils....`)

Reply:
Okk

Reply:
Should I do the same for Fast jar test too ?

Reply:
Sure